guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -154,13 +157,14 @@ protected int nextEscapeIndex(CharSequence csq, int start, int end) {
    * @throws NullPointerException if {@code string} is null
    * @throws IllegalArgumentException if invalid surrogate characters are encountered
    */
-  protected final String escapeSlow(String s, int index) {
+  @SuppressWarnings(value = {"assignment.type.incompatible", "argument.type.incompatible"})//for now
+  protected final String escapeSlow(String s, @IndexOrHigh("#1") int index) {
     int end = s.length();
 
     // Get a destination buffer and setup some loop variables.
     char[] dest = Platform.charBufferFromThreadLocal();
-    int destIndex = 0;
-    int unescapedChunkStart = 0;
+    @NonNegative @LTLengthOf(value = "dest", offset = "(index - unescapedChunkStart) - 1") int destIndex = 0;
```

Comment:
A I have commented in a google doc:
The internal expression parser is currently limited and cannot parse expressions with parentheses like this. In this case, they are superfluous anyway.
Moreover, you cannot refer to a variable before it is declared

Reply:
I'll fix it right away.

Reply:
@NhatDinh, is there any problem with this?